src/NodaTime/Calendars/WondrousYearMonthDayCalculator.cs

```diff
@@ -0,0 +1,333 @@
+
+            var firstDay = gregorianStart + nawRuz.DayOfYear - 1;
+
+            return firstDay;
+        }
+
+        /// <summary>
+        /// Test this possible date
+        /// </summary>
+        /// <param name="year"></param>
+        /// <param name="month">Ayyam-i-Ha is appended to month 18</param>
+        /// <param name="day"></param>
+        internal override void ValidateYearMonthDay(int year, int month, int day)
+        {
+            Preconditions.CheckArgumentRange(nameof(year), year, WondrousMinYear, WondrousMaxYear);
+
+            switch (month)
+            {
+                case AyyamiHaMonth0:
+                    Preconditions.CheckArgumentRange(nameof(day), day, 1, DaysInAyyamHa(year));
+                    break;
+                case Month18:
+                    // allow Ayyam-i-Ha days in this month
+                    Preconditions.CheckArgumentRange(nameof(day), day, 1, DaysInMonth + DaysInAyyamHa(year));
+                    break;
+                default:
+                    Preconditions.CheckArgumentRange(nameof(month), month, 1, MonthsInYear);
+                    Preconditions.CheckArgumentRange(nameof(day), day, 1, DaysInMonth);
+                    break;
+            }
+        }
+
+        /// <param name="year"></param>
+        /// <remarks>All years have 19 months</remarks>
+        /// <remarks>Tested</remarks>
+        /// <returns></returns>
+        internal override int GetMonthsInYear(int year)
+        {
+            return MonthsInYear;
+        }
+
+        /// <param name="year"></param>
+        /// <param name="month"></param>
+        /// <remarks>All months have 19 days. Ayyam-i-Ha usually has 4 or 5 days.</remarks>
+        /// <returns></returns>
+        internal override int GetDaysInMonth(int year, int month)
+        {
+            Preconditions.CheckArgumentRange(nameof(year), year, WondrousMinYear, WondrousMaxYear);
+
+            var daysInMonth = DaysInMonth;
+
+            if (month == AyyamiHaMonth0)
+            {
+                daysInMonth = WondrousCalendarHelper.GetYearInfo(year).DaysInAyyamiHa;
+            }
+
+            return daysInMonth;
+        }
+
+        internal override bool IsLeapYear(int year)
+        {
+            var isLeapYear = WondrousCalendarHelper.GetYearInfo(year).DaysInAyyamiHa != 4;
+
+            return isLeapYear;
+        }
+
+        internal bool IsInAyyamiHa(YearMonthDay ymd)
+        {
+            var isInAyyamiHa = ymd.Month == Month18 && ymd.Day > DaysInMonth;
+
+            return isInAyyamiHa;
+        }
+
+        /// <param name="start"></param>
+        /// <param name="months"></param>
+        /// <remarks>
+        ///     Adding months around Ayyam-i-Ha is not well defined. Because they are "days outside of time", this
+        ///     implementation will skip over them.
+        ///     Can never get to Ayyam-i-Ha with this. From Month 18, will go to month 19.
```

Comment:
Although presumably you can get from Ayyam-i-Ha in one year to Ayyam-i-Ha in another year.